### ISSUE TITLE
Telegram Alerts

### ISSUE THREAD

**Issue by online-ua** [Xtrendence/Cryptofolio#39]
Get notified when a coin goes above or below a price target.

**Comment by Xtrendence** [comment]
I'll look into this and see if local notifications are possible to trigger through a scheduled task (e.g. a background task that checks the price every 15m, and sends an alert if the price is above or below a target). 

In general, this would be easy to add if the API were a Node.js one, but with PHP it'd be a bit difficult, and it's even harder since the whole thing is self-hosted. The way notifications work on mobile is through Firebase mostly. 

Since you mentioned it working through Telegram, it might be easier to do. So like a background task for the mobile app that sends a request to the Cryptofolio API, and the API checks the price, and then sends a Telegram notification if appropriate. However, this would need you to create a Telegram bot and give Cryptofolio the API key and whatnot, since the main goal of the project is to not have a centralized server or any features that call home or use a common service/server.

**So in short**: Local notifications from the mobile app itself are most likely possible, but the Cryptofolio API or Telegram cannot be the ones doing this without having to connect to Google or require further work to set up on the users' part.

If that's okay with you, I'll add local notifications so you still get notified when the price reaches a target, but Telegram alerts or notifications originating from the API unfortunately won't be possible while staying within the scope of the project and not utilizing external services.

**Comment by Xtrendence** [comment]
I've added this feature to the [project board](https://github.com/Xtrendence/Cryptofolio/projects/1#card-63626600), and I'll try to include it with V.2.2 if I can finish developing and implementing the other features I had planned for that release in time. Thank you for the suggestion by the way! :)
